File: frontend/fundrbolt-admin/src/lib/api/quick-sale.ts
/**
 * Quick Sale API Client
 *
 * Provides functions for quick ticket sales at check-in.
 */
import apiClient from '@/lib/axios'

// ================================
// Types
// ================================

export interface QuickSaleGuestInfo {
  name: string
  email?: string | null
  phone?: string | null
}

export interface QuickSaleRequest {
  ticket_package_id: string
  quantity: number
  buyer_name: string
  buyer_email: string
  buyer_phone?: string | null

  // Address fields
  address_line1?: string | null
  address_line2?: string | null
  city?: string | null
  state?: string | null
  postal_code?: string | null
  country?: string | null

  guests: QuickSaleGuestInfo[]
  payment_method?: string

  // Payment details
  card_last_four?: string | null
  check_number?: string | null

  // Bidder and table assignment (null = auto-assign)
  bidder_number?: number | null
  table_number?: number | null

  check_in_immediately?: boolean
  notes?: string | null
}

export interface QuickSaleGuestResult {
  id: string
  registration_id: string
  name: string
  email: string | null
  phone: string | null
  is_primary: boolean
  checked_in: boolean
  bidder_number: number | null
  table_number: number | null
}

export interface QuickSaleResponse {
  purchase_id: string
  registration_id: string
  confirmation_code: string
  ticket_count: number
  package_name: string
  total_amount: number
  payment_method: string
  guests: QuickSaleGuestResult[]
  message: string
}

// ================================
// API Functions
// ================================

/**
 * Create a quick ticket sale at check-in.
 */
export async function createQuickSale(
  eventId: string,
  request: QuickSaleRequest
): Promise<QuickSaleResponse> {
  const response = await apiClient.post<QuickSaleResponse>(
    `/admin/events/${eventId}/quick-sale`,
    request
  )
  return response.data
}

File: backend/app/schemas/quick_sale.py
"""Pydantic schemas for quick sale at check-in."""

import uuid
from decimal import Decimal

from pydantic import BaseModel, EmailStr, Field


class QuickSaleGuestInfo(BaseModel):
    """Guest information for quick sale."""

    name: str = Field(..., min_length=1, max_length=255, description="Guest's full name")
    email: EmailStr | None = Field(None, description="Guest's email address (optional)")
    phone: str | None = Field(None, max_length=20, description="Guest's phone number (optional)")


class QuickSaleRequest(BaseModel):
    """Request schema for quick ticket sale at check-in."""

    ticket_package_id: uuid.UUID = Field(..., description="ID of the ticket package to purchase")
    quantity: int = Field(default=1, ge=1, le=20, description="Number of tickets to purchase")
    buyer_name: str = Field(..., min_length=1, max_length=255, description="Buyer's full name")
    buyer_email: EmailStr = Field(..., description="Buyer's email address")
    buyer_phone: str | None = Field(None, max_length=20, description="Buyer's phone number")
    guests: list[QuickSaleGuestInfo] = Field(
        default_factory=list,
        description="Additional guests (buyer is automatically the first guest)",
    )
    payment_method: str = Field(
        default="cash", description="Payment method: cash, check, card, other"
    )
    check_in_immediately: bool = Field(
        default=True, description="Check in all guests immediately after purchase"
    )
    notes: str | None = Field(None, max_length=500, description="Optional notes for this sale")


class QuickSaleGuestResult(BaseModel):
    """Result for a single guest in the quick sale."""

    id: uuid.UUID
    registration_id: uuid.UUID
    name: str
    email: str | None
    phone: str | None
    is_primary: bool
    checked_in: bool
    bidder_number: int | None = None

    class Config:
        from_attributes = True


class QuickSaleResponse(BaseModel):
    """Response schema for quick sale creation."""

    purchase_id: uuid.UUID
    registration_id: uuid.UUID
    confirmation_code: str
    ticket_count: int
    package_name: str
    total_amount: Decimal
    payment_method: str
    guests: list[QuickSaleGuestResult]
    message: str

    class Config:
        from_attributes = True
